### PR TITLE
Auditing the dependencies and syncing `dependencies.yaml` with `meta.yaml`

### PR DESCRIPTION
**Pull request by mdemoret-nv**
## Description

In Progress

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

**Review by dagardner-nv — APPROVED** [review]

File: dependencies.yaml [hunk 1]
```diff
@@ -165,32 +195,82 @@ channels:
...
-          - pkg-config=0.29 # for mrc cmake
-          - protobuf=4.24
           - pybind11-stubgen=0.10.5
           - rapidjson=1.1.0
+          - rdma-core>=48 # Needed for DOCA.
           - scikit-build=0.17.6
-          - sysroot_linux-64=2.17
-          - tritonclient=2.34
-          - ucx=1.15
+          - versioneer-518
```
Comment (dagardner-nv) [inline dependencies.yaml:273]: I think the switch from `versioneer` to `versioneer-518` is causing build failures

File: dependencies.yaml [hunk 2]
```diff
@@ -228,31 +306,50 @@ dependencies:
...
+          - numpydoc=1.5
           - nvtabular=23.08.00
           - pydantic
+          # - python ##
           - python-confluent-kafka>=1.9.2,<1.10.0a0
           - python-graphviz
           - pytorch-cuda
           - pytorch=*=*cuda*
+          - pluggy=1.3
+          - pyarrow=*=*_cuda # Ensure we get a CUDA build. Version determined by cuDF
```
Comment (mdemoret-nv) [inline dependencies.yaml:345]: Remove this and default to cuDF

File: ci/conda/recipes/morpheus/meta.yaml [hunk 3]
```diff
@@ -58,53 +58,61 @@ outputs:
...
         - mlflow>=2.10.0,<3
         - mrc
-        - networkx>=2.8
+        - networkx=2.8.8
         - numpydoc =1.5.*
         - nvtabular =23.08.00
-        - pandas
+        - pydantic
         - pluggy =1.3.*
         - pyarrow * *_cuda # Ensure we get a CUDA build. Version determined by cuDF
```
Comment (mdemoret-nv) [inline ci/conda/recipes/morpheus/meta.yaml:107]: Remove pyarrow and defer to cudf here too

**Pull request merged**